File: Java/Q39.java
import java.util.Scanner;

public class OctalDecimal {
    static int octalToDecimal(int octal) {
        int decimal = 0, i = 0;
        while (octal != 0) {
            int rem = octal % 10;
            octal /= 10;
            decimal += rem * Math.pow(8, i);
            i++;
        }
        return decimal;
    }

    static String decimalToOctal(int decimal) {
        return Integer.toOctalString(decimal);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter octal number: ");
        int octal = sc.nextInt();
        System.out.println("Decimal: " + octalToDecimal(octal));

        System.out.print("Enter decimal number: ");
        int decimal = sc.nextInt();
        System.out.println("Octal: " + decimalToOctal(decimal));
        sc.close();
    }
}

File: C/Q39.c
#include <stdio.h>
#include <math.h>

// Function to convert octal to decimal
int octalToDecimal(int octal) {
    int decimal = 0, i = 0;
    while (octal != 0) {
        int digit = octal % 10;
        decimal += digit * pow(8, i);
        octal /= 10;
        i++;
    }
    return decimal;
}

// Function to convert decimal to octal
void decimalToOctal(int decimal) {
    int octal[32], i = 0;
    while (decimal > 0) {
        octal[i++] = decimal % 8;
        decimal /= 8;
    }
    printf("Octal: ");
    for (int j = i - 1; j >= 0; j--)
        printf("%d", octal[j]);
    printf("\n");
}

int main() {
    int choice;
    printf("Choose conversion:\n1. Octal to Decimal\n2. Decimal to Octal\n");
    scanf("%d", &choice);

    if (choice == 1) {
        int octal;
        printf("Enter octal number: ");
        scanf("%d", &octal);
        printf("Decimal: %d\n", octalToDecimal(octal));
    } else if (choice == 2) {
        int decimal;
        printf("Enter decimal number: ");
        scanf("%d", &decimal);
        decimalToOctal(decimal);
    } else {
        printf("Invalid choice\n");
    }

    return 0;
}
